### PR TITLE
feat: Added features document.  Some corrections to components presence

### LINKED ISSUE #1661: Add documentation on logic for GTFS components
### Describe the problem

As GTFS components evolve, we want to provide clear public documentation on how we're defining the "presence" of a GTFS component. This will help us make decisions in the future if we want more granular calculations (e.g tracking the actual usage of a component in the feed). [The code is here.](https://github.com/MobilityData/gtfs-validator/blob/master/main/src/main/java/org/mobilitydata/gtfsvalidator/report/model/FeedMetadata.java).

### Proposed solution

- A 2 column table where each component has its associated logic.

### Alternatives you've considered

- Programmatically generating this (may be too time consuming for the value at the moment) 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jcpitre**
**Summary:**

Closes #1661 

Added a document with the list of components-features and how we determine if they are present.
Removed Blocks component
Corrected the code where components were duplicated because of a typo in their name,

**Expected behavior:** 
I expect no change of behavior.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Add or update any needed [documentation](https://github.com/MobilityData/gtfs-validator/tree/master/docs) to the repo 
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Review by emmambd — COMMENTED** [review]
@jcpitre This looks good! I think it would be good to link to [the public GTFS components document](https://docs.google.com/spreadsheets/d/1kpbKOzlHtsJIPo3B4ABYu-Nxwvqutjcdcvgpa0wZvtA/edit#gid=1997312156) in the "list of components" text at the top.

File: docs/features.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Discovery of component in feeds
+
+The validator will list the list of components that it finds in the processed feed according to this table:
+
+| Component                | Feature                   | How is the presence of a featuree determined (minimum requirements)                                                                                                                                                                                                                                                                                                                                                                                |
```
Comment (emmambd) [inline docs/features.md:5]: typo: featuree should be feature

**Comment by jcpitre** [comment]
> @jcpitre This looks good! I think it would be good to link to [the public GTFS components document](https://docs.google.com/spreadsheets/d/1kpbKOzlHtsJIPo3B4ABYu-Nxwvqutjcdcvgpa0wZvtA/edit#gid=1997312156) in the "list of components" text at the top.

I thought [the public GTFS components document](https://docs.google.com/spreadsheets/d/1kpbKOzlHtsJIPo3B4ABYu-Nxwvqutjcdcvgpa0wZvtA/edit#gid=1997312156) was more a work document that was not that permanent.

**Comment by emmambd** [comment]
> > @jcpitre This looks good! I think it would be good to link to [the public GTFS components document](https://docs.google.com/spreadsheets/d/1kpbKOzlHtsJIPo3B4ABYu-Nxwvqutjcdcvgpa0wZvtA/edit#gid=1997312156) in the "list of components" text at the top.
> 
> I thought [the public GTFS components document](https://docs.google.com/spreadsheets/d/1kpbKOzlHtsJIPo3B4ABYu-Nxwvqutjcdcvgpa0wZvtA/edit#gid=1997312156) was more a work document that was not that permanent.

It's being used in public facing meetings, so we can go ahead and add it. It's important context for anyone who reads the documentation and doesn't know what components are.

**Comment by jcpitre** [comment]
> It's being used in public facing meetings, so we can go ahead and add it. It's important context for anyone who reads the documentation and doesn't know what components are.

Added the link

**Review by qcdyx — APPROVED** [review]
LGTM

**Review by emmambd — APPROVED** [review]
LGTM

**Pull request merged**